commands/provider/core_provider.go

```diff
@@ -32,38 +31,30 @@ func (p *CoreProvider) Start() error {
 
 // Stop stops core daemon and all running services.
 func (p *CoreProvider) Stop() error {
-	var wg sync.WaitGroup
-
 	ids, err := service.ListRunning()
 	if err != nil {
 		return err
 	}
 
-	var errC = make(chan error, len(ids))
-	wg.Add(len(ids))
+	var (
+		idsLen = len(ids)
+		errC   = make(chan error, idsLen)
+	)
 
 	for _, id := range ids {
 		go func(id string) {
-			defer wg.Done()
-			if _, err := p.client.StopService(context.Background(), &coreapi.StopServiceRequest{
+			_, err := p.client.StopService(context.Background(), &coreapi.StopServiceRequest{
 				ServiceID: id,
-			}); err == nil {
```

Comment:
This was definitely a bug, we add the err (which is nil in the `errC`) I guess it's related to what @krhubert was talking about with the nil in the error channels but I don't quite understand what was the point. If you can explain a bit more that would be great.